planner/core/find_best_task.go

```diff
@@ -289,6 +289,12 @@ func (p *baseLogicalPlan) enumeratePhysicalPlans4Task(physicalPlans []PhysicalPl
 
 // compareTaskCost compares cost of curTask and bestTask and returns whether curTask's cost is smaller than bestTask's.
 func compareTaskCost(ctx sessionctx.Context, curTask, bestTask task) (curIsBetter bool, err error) {
+	if curTask.invalid() {
+		return false, nil
+	}
+	if bestTask.invalid() && !curTask.invalid() {
```

Comment:
The first IF has already checked whether `curTask` is invalid so here `curTask` must be valid.

Reply:
Updated.